File: java/t/test/ImmutableObject/Builder.java
package test.ImmutableObject;

import static org.junit.Assert.assertEquals;

import org.junit.Test;
import org.junit.Ignore;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import common.*;
import ImmutableObject.*;

public class Builder {

   public String CreateTestOutput( SoupBuffetBuilder sbb ) { 
		ImmutableObject.SoupBuffet sb = sbb.buildSoupBuffet();
		return sb.getSoupBuffetName() + sb.getTodaysSoups();
   }     
   
	@Test 
   public void thetester() {
       
       org.junit.Assert.assertEquals("Boston soup buffet builder test",
			"At the Boston Soup Buffet Today's Soups!   Chicken Soup: ChickenSoup Clam Chowder: QuahogChowder Fish Chowder: ScrodFishChowder Minnestrone: Minestrone Pasta Fazul: Pasta Fazul Tofu Soup: Tofu Soup Vegetable Soup: Vegetable Soup",
			"At the " + CreateTestOutput( new BostonSoupBuffetBuilder())
			);

       org.junit.Assert.assertEquals("Honolulu soup buffet builder test",
			"At the Honolulu Soup Buffet Today's Soups!   Chicken Soup: ChickenSoup Clam Chowder: PacificClamChowder Fish Chowder: OpakapakaFishChowder Minnestrone: Minestrone Pasta Fazul: Pasta Fazul Tofu Soup: Tofu Soup Vegetable Soup: Vegetable Soup",
			"At the " + CreateTestOutput( new HonoluluSoupBuffetBuilder() )
		);
   }
}      

File: clojure/src/patterns/creational/builder.clj
(ns patterns.creational.builder
   (:require
      [common.soups :refer :all]
      [common.soupbuffet :refer :all]
      [patterns.creational.abstract_factory :refer :all])
	(:import (common.soupbuffet soupbuffet))
)

(defn soupbuffetbuilder-helper [bldr name]
	(soupbuffet. name
		((:make-chickensoup bldr))
		((:make-clamchowder bldr))
		((:make-fishchowder bldr))
		((:make-minnestrone bldr))
		((:make-pastafazul bldr)) 
		((:make-tofusoup bldr))
		((:make-vegetablesoup bldr))
	))

(defn bostonsoupbuffetbuilder []
	(soupbuffetbuilder-helper (bostonsoupfactory) "Boston Soup Buffet"))

(defn honolulusoupbuffetbuilder []
	(soupbuffetbuilder-helper (honolulusoupfactory) "Honolulu Soup Buffet"))
	
